### ISSUE TITLE
Proposal: Add placeholders for blocked images?

### ISSUE THREAD

**Issue by amiv1** [uBlock-LLC/uBlock#1146]
I love the power of this extension and got a proposal to make it even better.

I'm using an "advanced user" mode so I can manually choose which third-party JavaScript should be executed and I think blocking third-party images as well would even more improve privacy when surfing the internet, since there are tracking mechanisms by hotlinking images on a website. Though uBlock provides this functionality, it's not always obvious that an image has been blocked on a page. It would be great not to just block third-party images but also put a placeholder instead of it so user knows that something is intended to be there, and temporarily loading an image by clicking on a placeholder would be the gold feature!

Thank you for the great project!

**Comment by Mikey1993** [comment]
The feature of 3rd party images seems like an awesome idea to enhance privacy.
+1

**Comment by harshanvn** [comment]
Yes, I like the idea of blocking 3 Party images. It will foil web bugs too :)

**Comment by lewisje** [comment]
To be clear, uBlock _already_ blocks third-party images, but at the moment it does so by cancelling the HTTP request; the proposal is to put a placeholder in instead, and I think that should only be done if the user opts into it, because pages look cleaner if the blocked content just isn't there at all.

BTW I originally preferred replacing blocked images with 1x1 transparent placeholders, and similar minimal-content replacement for other blocked content, but that doesn't play nicely with HTTPS Everywhere and other extensions that rely on redirection via the webRequest API, as gorhill himself discovered early on: https://github.com/gorhill/uBlock/issues/18

Maybe if declarativeWebRequest ever goes stable, uBlock for Chrome could redirect blocked images to a standard transparent image, and blocked scripts, stylesheets, and subdocuments to a standard empty document, without causing errors in other extensions, but that's been on hold for a long time: https://developer.chrome.com/extensions/declarativeWebRequest

**Comment by gorhill** [comment]
> 3rd party images seems like an awesome idea to enhance privacy

What is it with the this common habit of piling more unrelated issues into a very specific issue? The issue opened here is about "placeholders for blocked images".

**Comment by gorhill** [comment]
Another thread where the "placeholder" topic was covered: #869.

**Comment by gorhill** [comment]
Ok, example of a page where Firefox's/Chromium's default placeholders are no good: 

http://www.gamedev.net/page/resources/_/technical/game-programming/doom3-is-the-proof-that-keep-it-simple-works-r4009

Just blank spaces.

**Edit:** Duh, I had forgotten to disable "Hide placeholders of blocked elements". The placeholders show fine when the settings is un-checked.

So the missing part in OP's request is "temporarily loading an image by clicking on a placeholder"

**Comment by gorhill** [comment]
@lewisje If ever I change my mind about supporting custom placeholders, which is now likely given the above case, it won't be through redirection (because of the HTTPS-E problem), but through content scripts doing the replacement.

**Comment by Mikey1993** [comment]
> and I think blocking third-party images as well would even more improve privacy when surfing the internet
> 
> 3rd party images seems like an awesome idea to enhance privacy 
> 
> What is it with the this common habit of piling more unrelated issues into a very specific issue? The issue opened here is about "placeholders for blocked images".

Not unrelated, the OP has mentioned it.

**Comment by amiv1** [comment]
@Mikey1993, I said that uBlock provides image blocking in the very same post you're quoting this from. The context matters as well :)

**Comment by dstengele** [comment]
It would be nice to be able to customize the placeholder image. Google for example shows cat pictures instead of ads with their [Contributor program](http://www.droid-life.com/2015/07/28/google-contributor-now-open-to-public-support-websites-and-see-cat-ads/). This could also use a Service like [Placekitten](https://placekitten.com/) to provide custom sized cat pictures on the fly.
